ros2_control_demo_hardware/src/rrbot_system_quadruped.cpp

```diff
@@ -0,0 +1,320 @@
+{
+  RCLCPP_INFO(
+    rclcpp::get_logger("RRBotSystemQuadrupedHardware"),
+    "Starting ...please wait...");
+
+  for (int i = 0; i <= hw_start_sec_; i++) {
+    rclcpp::sleep_for(std::chrono::seconds(1));
+    RCLCPP_INFO(
+      rclcpp::get_logger("RRBotSystemQuadrupedHardware"),
+      "%.1f seconds left...", hw_start_sec_ - i);
+  }
+
+  // set some default values
+  for (uint i = 0; i < hw_states_.size(); i++) {
+    if (std::isnan(hw_states_[i].position)) {
+      hw_states_[i] = {0.0, 0.0, 0.0, 0.0, 0.0};
+      hw_commands_[i] = {0.0, 0.0, 0.0, 0.0, 0.0};
+    }
+  }
+
+  status_ = hardware_interface::status::STARTED;
+
+  RCLCPP_INFO(
+    rclcpp::get_logger("RRBotSystemQuadrupedHardware"),
+    "System Sucessfully started!");
+
+  return return_type::OK;
+}
+
+return_type RRBotSystemQuadrupedHardware::stop()
+{
+  RCLCPP_INFO(
+    rclcpp::get_logger("RRBotSystemQuadrupedHardware"),
+    "Stopping ...please wait...");
+
+  for (int i = 0; i <= hw_stop_sec_; i++) {
+    rclcpp::sleep_for(std::chrono::seconds(1));
+    RCLCPP_INFO(
+      rclcpp::get_logger("RRBotSystemQuadrupedHardware"),
+      "%.1f seconds left...", hw_stop_sec_ - i);
+  }
+
+  status_ = hardware_interface::status::STOPPED;
+
+  RCLCPP_INFO(
+    rclcpp::get_logger("RRBotSystemQuadrupedHardware"),
+    "System sucessfully stopped!");
+
+  return return_type::OK;
+}
+
+hardware_interface::return_type RRBotSystemQuadrupedHardware::read()
+{
+  RCLCPP_INFO(
+    rclcpp::get_logger("RRBotSystemQuadrupedHardware"),
+    "Reading...");
+
+  for (uint i = 0; i < hw_states_.size(); i++) {
+    // Simulate RRBotQuadruped's PD+ computation
+    hw_states_[i].effort = hw_commands_[i].effort +
+      hw_commands_[i].Kp*
+      (hw_states_[i].position -
+       hw_commands_[i].position) +
+      hw_commands_[i].Kd*
+      (hw_states_[i].velocity -
+       hw_commands_[i].velocity);
+
+    //
+    RCLCPP_INFO(
+      rclcpp::get_logger("RRBotSystemQuadrupedHardware"),
+      "Got state %.5f for joint %d!", hw_states_[i], i);
+  }
+  RCLCPP_INFO(
+    rclcpp::get_logger("RRBotSystemQuadrupedHardware"),
+    "Joints sucessfully read!");
+
+  return return_type::OK;
+}
+
+hardware_interface::return_type
+RRBotSystemQuadrupedHardware::write()
+{
+  RCLCPP_INFO(
+    rclcpp::get_logger("RRBotSystemQuadrupedHardware"),
+    "Writing...");
+
+  for (uint i = 0; i < hw_commands_.size(); i++) {
+    // Simulate sending commands to the hardware
+    RCLCPP_INFO(
+      rclcpp::get_logger("RRBotSystemQuadrupedHardware"),
+      "Got command %.5f for joint %d!", hw_commands_[i], i);
+  }
+  RCLCPP_INFO(
+    rclcpp::get_logger("RRBotSystemQuadrupedHardware"),
+    "Joints sucessfully written!");
+
+  return return_type::OK;
+}
+
+}  // namespace quadruped
+
+}  // namespace ros2_control_demo_hardware
+
+#include "pluginlib/class_list_macros.hpp"
+
+PLUGINLIB_EXPORT_CLASS(
+  ros2_control_demo_hardware::quadruped::RRBotSystemQuadrupedHardware,
```

Comment:
There is no need to introduce an additional namespace. Or are you planning to have multiple examples for this category?

```suggestion
  ros2_control_demo_hardware::RRBotSystemQuadrupedHardware,
```